File: 01gpio/03digital_in/ir/ir.js
var onoff = require('onoff');
var pin = new onoff.Gpio(18, 'in', 'both');
var val = -1;
while(true){
	var read = pin.readSync();
	if(read != val){
		val = read;
		console.log(val == 0 ? 'no IR': 'IR detected');
	}
}

File: 01gpio/03digital_in/ir/ir.py
import RPi.GPIO as GPIO
import time

PIN = 18
GPIO.setmode(GPIO.BCM)
GPIO.setup(PIN, GPIO.IN)
val = -1

try:
    while True:
        read = GPIO.input(PIN)
        if read != val:
            val = read
            print(val == 0 and 'no IR' or 'IR detected')
        #time.sleep(0.1)
finally:
    print("clean up.")
    GPIO.cleanup()
